pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -387,6 +387,26 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 
 		cdLog.Debug("cluster is already installed, no processing of provision needed")
 		r.cleanupInstallLogPVC(cd, cdLog)
+
+		if cd.Spec.ClusterMetadata.AdminKubeconfigSecretRef.Name != "" && (cd.Status.WebConsoleURL == "" || cd.Status.APIURL == "") {
+			adminKubeconfigSecret := &corev1.Secret{}
+			if err := r.Get(context.Background(), types.NamespacedName{Namespace: cd.Namespace, Name: cd.Spec.ClusterMetadata.AdminKubeconfigSecretRef.Name}, adminKubeconfigSecret); err != nil {
```

Comment:
Can we create a function to call to do this work that can be called from `reconcileCompletedProvision` as well rather than duplicating the code?